### PR TITLE
Add migration blog post

### PR DESCRIPTION
**Pull request by electrum**

### CODE REVIEW

**Review by findepi — COMMENTED** [review]
Stylistically LGTM.
You can still want to wait for someone to review content completeness, if you wish.

File: _posts/2021-01-04-migrating-from-prestosql-to-trino.md [hunk 1]
```diff
@@ -0,0 +1,134 @@
...
+name. For example, for the Hive connector:
+
+```
+jmx.base-name=presto.plugin.hive
+```
+
+# Thrift connector
+
+The [Thrift connector](/docs/current/connector/thrift.html) had many backwards
+incompatible changes to both the Thrift service interface and the configuration
```
Comment (dain) [inline _posts/2021-01-04-migrating-from-prestosql-to-trino.md:119]: maybe link to the release notes for the list of changes

File: _posts/2021-01-04-migrating-from-prestosql-to-trino.md [hunk 2]
```diff
@@ -0,0 +1,134 @@
...
+properties. You'll need update all of your implementations of the Thrift service
```
Comment (dain) [inline _posts/2021-01-04-migrating-from-prestosql-to-trino.md:120]: maybe drop the contraction

File: _posts/2021-01-04-migrating-from-prestosql-to-trino.md [hunk 3]
```diff
@@ -0,0 +1,134 @@
...
+
+<!--more-->
+
+# Things that haven't changed
+
+Let's start with the good news. For end users running queries against Trino,
+everything should be the same. There are no changes to the SQL language,
+SQL functions, session properties, etc.
+
+Users will now see *Trino* in error messages, and error stack traces will have
```
Comment (mosabua) [inline _posts/2021-01-04-migrating-from-prestosql-to-trino.md:26]: I would change to present tense.

Users now see *Trino* in error message ...

**Comment by mosabua** [comment]
I think all of this should go into the documentation either in addition or instead. Similar to how we have https://trino.io/docs/current/migration/from-hive.html

File: _posts/2021-01-04-migrating-from-prestosql-to-trino.md [hunk 4]
```diff
@@ -0,0 +1,134 @@
...
+
+As we previously announced, we're
+[rebranding Presto SQL as Trino]({% post_url 2020-12-27-announcing-trino %}).
+Now comes the hard part: migrating to the new version of the software.
+We just released the first version,
+[Trino 351](/docs/current/release/release-351.html),
+which uses the name Trino everywhere, both internally and externally.
+Unfortunately, there are various unvoidable compatibility aspects that
+administrators of Trino need to know about. We hope this post explains
+them well and makes the transition as smooth as possible.
```
Comment (mosabua) [inline _posts/2021-01-04-migrating-from-prestosql-to-trino.md:16]: them well, and makes

File: _posts/2021-01-04-migrating-from-prestosql-to-trino.md [hunk 5]
```diff
@@ -0,0 +1,134 @@
...
+a different package name, but otherwise they won't know that anything has changed.
+All of their views, reports, or other stored queries will work as before.
+
+Similarly for administrators, except for a few things noted in the
+[Trino 351 release notes](/docs/current/release/release-351.html),
+all the configuration properties are the same.
+
+# Client protocol compatiblity
+
+The client protocol is how clients like the
```
Comment (mosabua) [inline _posts/2021-01-04-migrating-from-prestosql-to-trino.md:36]: clients, like the

File: _posts/2021-01-04-migrating-from-prestosql-to-trino.md [hunk 6]
```diff
@@ -0,0 +1,134 @@
...
+[CLI](docs/current/installation/cli.html) or
+[JDBC driver](/docs/current/installation/jdbc.html)
+talk to Trino. It uses standard HTTP as the underlying communications
```
Comment (mosabua) [inline _posts/2021-01-04-migrating-from-prestosql-to-trino.md:39]: .. , talk to Trino.

File: _posts/2021-01-04-migrating-from-prestosql-to-trino.md [hunk 7]
```diff
@@ -0,0 +1,134 @@
...
+protocol, with some custom HTTP headers to communicate values
+to and from Trino. Unfortunately, those header names started with
+`X-Presto-` and thus had to be changed to `X-Trino-`.
```
Comment (mosabua) [inline _posts/2021-01-04-migrating-from-prestosql-to-trino.md:42]: comma and simplification

, and had to be changed

File: _posts/2021-01-04-migrating-from-prestosql-to-trino.md [hunk 8]
```diff
@@ -0,0 +1,134 @@
...
+
+The Trino CLI and JDBC driver send the new headers, so they are
+**only compatible with Trino versions 351 and newer**. Users should
+thus wait to upgrade the CLI or JDBC driver until the Trino servers
```
Comment (mosabua) [inline _posts/2021-01-04-migrating-from-prestosql-to-trino.md:46]: remove "thus"

File: _posts/2021-01-04-migrating-from-prestosql-to-trino.md [hunk 9]
```diff
@@ -0,0 +1,134 @@
...
+
+Out of the box, the Trino server does not work with older clients.
+However, in order to support a graceful transition, you can allow the
+server to support older clients by adding a configuration property:
+
+```
+protocol.v1.alternate-header-name=Presto
+```
+
+**We recommend using version 350 of CLI and JDBC driver as the transition version**,
```
Comment (mosabua) [inline _posts/2021-01-04-migrating-from-prestosql-to-trino.md:57]: End sentence and start new one (and use comma there)

versions. It has all the newest features, such as ..

File: _posts/2021-01-04-migrating-from-prestosql-to-trino.md [hunk 10]
```diff
@@ -0,0 +1,134 @@
...
+as it has all the newest features such as variable precision timestamps.
+It has been tested with a range of older server versions and is the last
```
Comment (mosabua) [inline _posts/2021-01-04-migrating-from-prestosql-to-trino.md:59]: versions, and

File: _posts/2021-01-04-migrating-from-prestosql-to-trino.md [hunk 11]
```diff
@@ -0,0 +1,134 @@
...
+version to support older servers.
+
+# JDBC Driver
```
Comment (mosabua) [inline _posts/2021-01-04-migrating-from-prestosql-to-trino.md:62]: JDBC driver

File: _posts/2021-01-04-migrating-from-prestosql-to-trino.md [hunk 12]
```diff
@@ -0,0 +1,134 @@
...
+
+
+The URL prefix for the JDBC driver now starts with `jdbc:trino:` instead
+of `jdbc:presto:`. This means that any client applications using the
+JDBC driver will need to update their connection configuration. The old
```
Comment (mosabua) [inline _posts/2021-01-04-migrating-from-prestosql-to-trino.md:67]: driver needs to update

**Comment by mosabua** [comment]
At the bottom of the post we should have a section about where to find more help (on slack...)

**Review by martint — APPROVED** [review]

**Pull request merged**